xray/segment.go

```diff
@@ -234,8 +232,7 @@ func NewSegmentFromHeader(ctx context.Context, name string, h *header.Header) (c
 // Close a segment.
 func (seg *Segment) Close(err error) {
 	seg.Lock()
-	defer seg.Unlock()
-	if seg.parent != nil {
+	if seg.parent != nil || seg.Type == "Subsegment" {
```

Comment:
Why we add a condition `seg.Type == "Subsegment"` here? How to trigger this one? I think the stream strategy happened after the segment getting closed. I feel this condition will never be triggered since we do assigned `Subsegment` type during streaming strategy.

Reply:
Good point. This check is unnecessary and I have removed it.